From c6ae8568b99785465d1461c8878cdee08ff44eae Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint@balintreczey.hu>
Date: Tue, 4 Nov 2014 00:13:21 +0100
Subject: [PATCH] linux: Check for Logind first, then *Kit-s with UPower when
 detecting PM framework

Logind is the most likely candidate to work nowadays and latest UPower does
not provide suspend/hibernate API.
---
 xbmc/powermanagement/PowerManager.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/xbmc/powermanagement/PowerManager.cpp
+++ b/xbmc/powermanagement/PowerManager.cpp
@@ -78,12 +78,12 @@
   m_instance = new CAndroidPowerSyscall();
 #elif defined(TARGET_POSIX)
 #if defined(HAS_DBUS)
-  if (CConsoleUPowerSyscall::HasConsoleKitAndUPower())
+  if (CLogindUPowerSyscall::HasLogind())
+    m_instance = new CLogindUPowerSyscall();
+  else if (CConsoleUPowerSyscall::HasConsoleKitAndUPower())
     m_instance = new CConsoleUPowerSyscall();
   else if (CConsoleDeviceKitPowerSyscall::HasDeviceConsoleKit())
     m_instance = new CConsoleDeviceKitPowerSyscall();
-  else if (CLogindUPowerSyscall::HasLogind())
-    m_instance = new CLogindUPowerSyscall();
   else if (CUPowerSyscall::HasUPower())
     m_instance = new CUPowerSyscall();
 #if defined(HAS_HAL)
